solutions/usaco-416.mdx

```diff
@@ -0,0 +1,86 @@
+---
+id: usaco-416
+source: USACO Silver 2014 March
+title: The Lazy Cow
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
+
+## Implementation
+
+**Time Complexity** $\mathcal{O}(K \cdot N ^ 2)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+Depending on how you implement this problem, Java code may have trouble 
+getting the last testcase even though the algorithm is correct.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class TheLazyCow {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("lazy");
+		int n = io.nextInt();
+		int k = io.nextInt();
+		int[][] matrix = new int[2 * n + 1][2 * n + 1];
+
+		// read input and store it rotated by 45 degrees
+		for (int i = 0; i < n; i++) {
+			for (int j = 0; j < n; j++) {
+				matrix[i + j][n - i + j - 1] = io.nextInt();
+			}
+		}
+
+		// the 45 degree rotation expands the size of the matrix
+		// it also includes the cells with half distance
+		int t = (n + 1) % 2;  // t is used to avoid non-lattice points
```

Comment:
quickly looking at this, it seems a bit unintuitive how t works - maybe add a comment explaining it

Reply:
i did add a comment, just avoiding the non integer x and y coordinates, I use this to be consistent with the official solution

Reply:
Yeah I just thought an explanation of the logic behind how t is used would be nice. The official solution is a bit unclear about this too (I might be missing something idk).

Reply:
this.